desc/equilibrium.py

```diff
@@ -265,6 +270,108 @@ def change_resolution(
         if N_grid is not None and N_grid != self.N_grid:
             self._N_grid = N_grid
 
+    @classmethod
+    def from_near_axis(
+        cls, na_eq, r=0.1, L=None, M=8, N=None, ntheta=None, spectral_indexing="ansi"
+    ):
+        """Initialize an Equilibrium from a near-axis solution.
+
+        Parameters
+        ----------
+        na_eq : Qsc or Qic
+            Near-axis solution generated by pyQSC or pyQIC.
+        r : float
+            Radius of the desired boundary surface (in meters).
+        L : int (optional)
+            Radial resolution. Default 2*M for `spectral_indexing`==fringe, else M
+        M : int (optional)
+            Poloidal resolution. Default is 8
+        N : int (optional)
+            Toroidal resolution. Default is the max resolution provided by na_eq.nphi
+        ntheta : int, optional
+            Number of poloidal grid points used in the conversion. Default 2*M+1
+        spectral_indexing : str (optional)
+            Type of Zernike indexing scheme to use. Default ``'ansi'``
+
+        Returns
+        -------
+        eq : Equilibrium
+            Equilibrium approximation of the near-axis solution.
+
+        """
+        try:
+            # default resolution parameters
+            if L is None:
+                if spectral_indexing == "ansi":
+                    L = M
+                elif spectral_indexing == "fringe":
+                    L = 2 * M
+            if N is None:
+                N = int((na_eq.nphi - 1) / 2)
+            if ntheta is None:
+                ntheta = 2 * M + 1
+
+            inputs = {}
+            inputs["Psi"] = np.pi * r ** 2 * na_eq.spsi * na_eq.Bbar
+            inputs["NFP"] = na_eq.nfp
+            inputs["L"] = L
+            inputs["M"] = M
+            inputs["N"] = N
+            inputs["sym"] = not na_eq.lasym
+            inputs["spectral_indexing "] = spectral_indexing
+            inputs["pressure"] = np.array(
+                [[0, -na_eq.p2 * r ** 2], [2, na_eq.p2 * r ** 2]]
+            )
+            inputs["iota"] = None
+            inputs["current"] = np.array([[2, 2 * np.pi / mu_0 * na_eq.I2 * r ** 2]])
+            inputs["axis"] = FourierRZCurve(
+                R_n=np.concatenate((np.flipud(na_eq.rs[1:]), na_eq.rc)),
+                Z_n=np.concatenate((np.flipud(na_eq.zs[1:]), na_eq.zc)),
+                NFP=na_eq.nfp,
+            )
+            inputs["surface"] = None
+        except AttributeError as e:
+            raise ValueError("Input must be a pyQSC or pyQIC solution.") from e
+
+        rho, _ = special.js_roots(L, 2, 2)
+        grid = LinearGrid(rho=rho, theta=ntheta, zeta=na_eq.nphi, NFP=na_eq.nfp)
```

Comment:
do we want to use the "ocs" concentric grid for fitting here? Or at least oversample by default?

Reply:
This is the radial spacing used in the quadrature grid. We don't necessarily need to oversample.

Reply:
why do we need the quad grid radial spacing here though? isn;'t the ocs supposed to be the optimal sampling grid for fitting zernike